sympy/solvers/bivariate.py

```diff
@@ -169,15 +170,16 @@ def _lambert(eq, x):
     # solution of the given Lambert equation is like
     # sol = -c/b + (a/d)*LambertW(arg, k),
     # where arg = d/(a*b)*exp((c*d-b*f)/a/b) and k in lambert_real_branches.
-    # arg can have values of p roots of exp(num/den)**(1/p)[exp((c*d-b*f)/a/b)]
-    # and will required to have all solutions of x**p - exp(num/den).
+    # arg can have values of p roots of exp((c*d-b*f)/a/b) represented
+    # as exp(num/den)**(1/p) will required to have all
+    # solutions of x**p - exp(num/den).
 
     # calculating args for LambertW
     num, den = ((c*d-b*f)/a/b).as_numer_denom()
     p, den = den.as_coeff_Mul()
-    exp_t = Dummy('exp_t')
-    args = [d/(a*b)*e if p != S.One else d/(a*b)*exp(num/den) \
-            for e in solve(exp_t**p - exp(num/den), exp_t)]
+    e = exp(num/den)
+    t = Dummy('t')
+    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
```

Comment:
I am using t here to not confuse the user with x the variable of the main equation.